inst/include/child_model_simulation.hpp

```diff
@@ -13,54 +12,57 @@ void run_child_ageing(int t,
                       const State<ModelVariant, real_type> &state_curr,
                       State<ModelVariant, real_type> &state_next,
                       IntermediateData<ModelVariant, real_type> &intermediate) {
-  const auto demog = pars.base.demography;
-  const auto cpars = pars.children.children;
   static_assert(ModelVariant::run_child_model,
                 "run_hiv_child_infections can only be called for model variants where run_child_model is true");
```

Comment:
all static asserts should be at the top, `pars.children.children` doesnt exist if we are not running the child model so this would've errored before the `static_assert`